packages/core/src/storages/request_queue.ts

```diff
@@ -396,6 +401,62 @@ export class RequestQueue {
         return new Request(requestOptions as unknown as RequestOptions);
     }
 
+    /**
+     * Lists the current requests in the queue head, and locks them for a certain amount of time.
+     *
+     * @remark This method is currently **experimental**, and may change without a major version bump.
+     *
+     * @param options Options for the list and lock operation.
+     * @returns Returns the list of requests on the queue head, as well as how long the lock lasts.
+     */
+    async experimental__listAndLockHead(options: ListAndLockOptions): Promise<ListAndLockHeadResult> {
+        ow(options, ow.object.exactShape({
+            limit: ow.optional.number.integer,
+            lockSecs: ow.number.integer,
+        }));
+
+        const { limit = 100, lockSecs } = options;
+
+        const results = await this.client.listAndLockHead({ limit, lockSecs });
+
+        return results;
+    }
+
+    /**
+     * Prolongs a request's lock on the queue head.
+     *
+     * @remark This method is currently **experimental**, and may change without a major version bump.
+     *
+     * @param id The request id to prolong the lock for.
+     * @param options The options when prolonging the lock.
+     */
+    async experimental__prolongRequestLick(id: string, options: ProlongRequestLockOptions): Promise<ProlongRequestLockResult> {
```

Comment:
typo `prolongRequestLick` :)

Reply:
Wooops 😂, that'll be fixed once I also drop the experimental__ prefix as Martin requested